src/libraries/System.Composition.Convention/src/System.Composition.Convention.csproj

```diff
@@ -21,7 +22,16 @@
     <Compile Include="$(CommonPath)System\Composition\Diagnostics\TraceWriter.cs"
              Link="Common\System\Composition\Diagnostics\TraceWriter.cs" />
   </ItemGroup>
+
   <ItemGroup>
     <ProjectReference Include="$(LibrariesProjectRoot)System.Composition.AttributedModel\src\System.Composition.AttributedModel.csproj" />
   </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
```

Comment:
For some reason I thought that pure-OOBs didn't have to do this, but instead we allowed them to use the ref-pack.  That's not the case?

Reply:
Nope, every .NETCoreApp tfm needs to specify their dependencies today. That's what I would like to change (vision doc) so that only NetCoreAppCurrent needs to specify them.

Reply:
I see.  Another option here would be to split on OOBs and construct the ref-pack first.  That'd avoid them needing granular references (project or otherwise) on any .NETCore framework and make them look more like customer projects.  Not related to this PR though.

Reply:
Don't get me wrong here, this isn't a technical limitation. The targeting pack is already constructed before references for OOBs are resolved. Projects can opt into not defining their granular reference via this property: https://github.com/dotnet/runtime/blob/fc4a42746d9cf65ad3db4074207069ab9c168bc9/src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj#L8-L9

Reply:
In the vision doc I was proposing that we always specify granular references for .NETCoreApp regardless of if it's a pure OOB and never specify granular references for previous .NETCoreApp versions.

Reply:
Were you thinking that the value you got out of that was that it can provide a minimal project graph that doesn't need to build every assembly in the shared framework, at the expense of complicating the project file a bit?